engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodNode.java

```diff
@@ -154,29 +174,81 @@ private boolean typeCanOverride(MethodRootNode node, EnsoContext ctx) {
-            && panic.getEigentype() != methodOwnerType
-            && warning.getEigentype() != methodOwnerType;
+        && builtins.nothing() != methodOwnerType
+        && any.getEigentype() != methodOwnerType
+        && panic.getEigentype() != methodOwnerType
+        && warning.getEigentype() != methodOwnerType;
   }
 
   @Specialization(
-          replaces = "doFunctionalDispatchCachedSymbol",
-          guards = {"typesLibrary.hasType(self)", "!typesLibrary.hasSpecialDispatch(self)"})
+      replaces = "doFunctionalDispatchCachedSymbol",
+      guards = {"typesLibrary.hasType(self)", "!typesLibrary.hasSpecialDispatch(self)"})
   Object doFunctionalDispatchUncachedSymbol(
-          VirtualFrame frame,
-          State state,
-          UnresolvedSymbol symbol,
-          Object self,
-          Object[] arguments,
-          @CachedLibrary(limit = "10") TypesLibrary typesLibrary,
-          @Cached MethodResolverNode methodResolverNode) {
+      VirtualFrame frame,
+      State state,
+      UnresolvedSymbol symbol,
+      Object self,
+      Object[] arguments,
+      @CachedLibrary(limit = "10") TypesLibrary typesLibrary,
+      @Cached MethodResolverNode methodResolverNode) {
     Type selfTpe = typesLibrary.getType(self);
     Function function = resolveFunction(symbol, selfTpe, methodResolverNode);
     if (function == null) {
       throw new PanicException(
-              EnsoContext.get(this).getBuiltins().error().makeNoSuchMethod(self, symbol), this);
+          EnsoContext.get(this).getBuiltins().error().makeNoSuchMethod(self, symbol), this);
+    }
+    var resolvedFuncArgCount = function.getSchema().getArgumentsCount();
+    CallArgumentInfo[] invokeFuncSchema = invokeFunctionNode.getSchema();
+    // Check if this is a static method call on Any
+    if (isAnyEigenType(selfTpe) && arguments.length == resolvedFuncArgCount - 1) {
+      // function is a static method on Any, so the first two arguments in `invokeFuncSchema`
+      // represent self arguments.
+      boolean selfArgSpecified = false;
+      if (invokeFuncSchema.length > 1) {
+        selfArgSpecified =
+            invokeFuncSchema[1].getName() != null && invokeFuncSchema[1].getName().equals("self");
+      }
+
+      if (selfArgSpecified) {
+        // If there is a self named argument in the method call, we fall back to the old
+        // behavior - there will be no prepended Any.type self argument
+        assert arguments.length == invokeFunctionNode.getSchema().length;
+        return invokeFunctionNode.execute(function, frame, state, arguments);
+      }
+
+      Object[] argsWithPrependedSelf = new Object[arguments.length + 1];
+      argsWithPrependedSelf[0] = EnsoContext.get(this).getBuiltins().any().getEigentype();
+      System.arraycopy(arguments, 0, argsWithPrependedSelf, 1, arguments.length);
+
+      if (invokeAnyStaticFunctionNode == null) {
+        CompilerDirectives.transferToInterpreter();
```

Comment:
I think you want to also _invalidate_. Otherwise a compiled version of this method might remain in the JVM without noticing a new node `insert`ed...

Reply:
I also think it is a better idea. Done in 45d7fa55777b6cb317dcee57ca653261c56f1888